constraint/pkg/types/validation.go

```diff
@@ -9,6 +9,27 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+type EvaluationEngineType string
+
+const (
+	Undefined  EvaluationEngineType = "undefined"
+	RegoEngine EvaluationEngineType = "rego"
+)
+
+type ReviewMeta struct {
+	// evaluationLatency is the number of milliseconds it took to server a client.Review() call.
+	evaluationLatency float64
```

Comment:
I think `TotalTemplateRunTime` can box us into not easily using the `ReviewMeta` to show the run time per constraint.

`evaluationRunTime` ?

Reply:
The idea behind `GetStatsString()` is that the reporting is amorphous, which frees us from needing to figure out these details.

That aside `TotalTemplateRunTime` would actually free us up to report per-constraint metrics later. I'm assuming you are choosing the name `evaluationRunTime` because the scope is ambiguous, which would leave you room to change the meaning of the number later. Thing is, that would avoid breaking type checking for users who relied on that code, but change the meaning of the value, which could break the assumptions of whatever is built on top of it.

TL;DR:

* We should consider GetStatsString() the interface, and probably make these other members private
* Backwards compatibility is as much about the meaning of a field staying content as its name

Reply:
Thanks for the input here! I reworked the code per the guidance.